apps/bors_database/lib/context/permission.ex

```diff
@@ -0,0 +1,28 @@
+defmodule BorsNG.Database.Context.Permission do
+  @moduledoc """
+  The connection between a project and its reviewers.
+
+  People with this link can bring up the dashboard page and settings
+  for a project, and can r+ a commit. Otherwise, they can't.
+  """
+
+  use BorsNG.Database.Context
+
+  def user_has_permission_to_approve_patch(user, patch) do
+    %User{id: user_id} = user
+    %Patch{id: patch_id, project_id: project_id} = patch
+    delegated = UserPatchDelegation
+    |> where([d], d.user_id == ^user_id and d.patch_id == ^patch_id)
+    |> Repo.one()
```

Comment:
if you do this change, I guess it makes sense to do it in the opposite order `LinkUserProject` first, `UserPatchDelegation` second.